R/check_metadata_file_location.R

```diff
@@ -0,0 +1,27 @@
+#' Check that the metadata file is being submitted to the correct folder
+#'
+#' @inherit check_metadata_file_exists params
+#' @inherit check_tbl_col_types return
+#'
+#' @export
+check_metadata_file_location <- function(file_path) {
+  dir_name <- dirname(file_path)
+  check <- dir_name == "."
```

Comment:
The idea is that the model metadata file should be submitted to the hub's `model-metadata` folder, not a sub-folder within that directory. My thinking (outlined in [this discussion item](https://github.com/Infectious-Disease-Modeling-Hubs/hubValidations/discussions/23#discussion-5601085)) was that the `file_path` that is handed to this function is the path to a file relative to the `model-metadata` directory. In this check, we want to make sure the file is located directly within `model-metadata` and not in some further sub-directory. In that case, `dirname(file_path)` will be `"."`. (But if you have other ideas about how any of this should be set up, I'm very open to them!)

Reply:
...now that I'm thinking about it, maybe easier to just check in another function that the `file_path` is exactly `<model_id>.yml` with nothing leading it.  The existence of this function is maybe an artifact of my initial confusion about what the `file_path` would be relative to.  I initially was thinking the file path would be relative to the hub's root, then changed my mind about what the `file_path` would be relative to and updated this function to match that without really revisiting whether it would be necessary at all.

Reply:
Right, I also wasn't totally clear about this but basically landed in a similar place to where you did:

- Any file in (a subfolder of) `model-output` would have `validate_model_file` run on it, which also checks for the existence of a corresponding model metadata file but doesn't actually validate that model metadata file's contents. For these checks, the `file_path` is relative to `model-output`.
- Similarly, any file in (a subfolder of??) `model-metadata` would have `validate_model_metadata` run on in.
    - Here, (a subfolder of) is unnecessary because we could already reject these files on the grounds that they're in the wrong place... or we can just pass them on to `validate_model_metadata` and let this check verify that they're in the right place
- So, what to do about other files?
    - The strategy we use for the covid hub is that any file in another location is not validated by either of these functions, but a message pops up on the pull request saying something like, "this pull request includes one or more files that are in a location other than `model-output` or `model-metadata`". This strategy hasn't always been great (we've had an occasional accidental merge of files in unexpected locations because the PR validations didn't fail).
    - So, yes, another idea could be to make some kind of effort to detect whether or not each file that's part of a pull request is an output file or a metadata file and run the relevant checks on them... but that seems potentially tricky (not every yaml or csv file in a repo is one of those two)